include/envoy/stats/stats.h

```diff
@@ -167,6 +172,21 @@ class Histogram : public virtual Metric {
    * Records an unsigned value. If a timer, values are in units of milliseconds.
    */
   virtual void recordValue(uint64_t value) PURE;
+
+  /**
+   * Merges the histogram values collected during the flush interval.
+   */
+  virtual void merge() PURE;
+
+  /**
+   * Returns the Histogram Summary Statistics for the flush interval.
+   */
+  virtual HistogramStatistics intervalStatistics() const PURE;
```

Comment:
I'm not actually sure how this compiles. I think it's luck if `HistogramStatistics` is returned by value. We either need to define `HistogramStatistics` in this file, or I think probably more optimally, create an abstract interface for `HistogramStatistics` and return by const ref. I'm not sure which one will be easier. (The question on whether we can return by const ref is not really dependent on whether we use an abstract interface or not.)